packages/backend/src/graphql/queries/test-connection.ts

```diff
@@ -10,12 +11,13 @@ const testConnection: QueryResolvers['testConnection'] = async (
   params,
   context,
 ) => {
-  let connection = await context.currentUser
-    .withAccessibleConnections({ requiredRole: 'viewer' })
-    .findOne({
-      'connections.id': params.connectionId,
-    })
-    .throwIfNotFound()
+  let connection = await getConnection({
```

Comment:
if flowId is undefined, only fetch from user's connections. 

if flowId is defined, attempt to fetch from both